### PR TITLE
chore: move to vitest

### PR DESCRIPTION
**Pull request by westeezy**
# Tasks:

## Update  Node setup
* turn off `save=false` in `.npmrc` to avoid annoyances
* add `.nvmrc` and point to v16

## Migrate from Karma -> Vitest
* Removed `sync-browser-mocks` in favor of a simple global stub
* Move tests to vitest runner
* Colocate test in `/src/__tests__`

## Move Diagrams to Mermaid
* Moving diagrams from images to mermaid for easier editing later

## Update License
* Match Apache 2 License we use elsewhere

### CODE REVIEW

**Comment by westeezy** [comment]
@jamischarles you got me thinking about `beaver-logger` more so making some updates. Overall plan is to increase test coverage + modernize. We should chat some more soon.

cc @sioked

File: src/__tests__/test.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 /* @flow */
+import { describe, it } from "vitest";
 
 import {
```
Comment (westeezy) [inline src/__tests__/test.js:4]: working on this file now so CI will fail. Just noticed it after hastily opening PR

Comment (westeezy) [reply src/__tests__/test.js]: done

File: package.json [hunk 2]
```diff
@@ -95,20 +95,24 @@
...
+    "@vitest/ui": "^0.29.2",
     "cross-env": "^7.0.3",
     "flow-bin": "0.135.0",
     "flow-typed": "^3.8.0",
     "husky": "^7.0.4",
     "jest": "^29.3.1",
+    "jsdom": "^21.1.1",
     "lint-staged": "^12.4.0",
-    "mocha": "^4.0.0",
+    "msw": "^1.1.0",
```
Comment (dtjones404) [inline package.json:111]: Is Vitest using this?  I don't see it imported anywhere in the code.

Comment (westeezy) [reply package.json:111]: good catch, i tried it for the mocks but ended up not using it. Will remove now

**Review by jamischarles — APPROVED** [review]
Looks great! Thx @westeezy for bringing the heathens the gospel of vitest.

**Review by jamischarles — APPROVED** [review]

**Review by dtjones404 — APPROVED** [review]

**Pull request merged**